### PR TITLE
Client & Job Timelines

### LINKED ISSUE #112: Add Timeline to Client
Add timeline collection for Client. We need it for audits (i.e. the Individual Client Export)

Also add in frontend timeline stuff.

### PR DESCRIPTION
**Pull request by helen-li**
# Description

Fixes #112 
* No new dependencies added
* Added timelines to the client and job home pages (the screenshots are in this order too)
* Added the timeline endpoints to the client collection (sample curl request and response below)
* Modified any pre-existing timeline-related code to dynamically adapt for more different categories 
* Fixed a tiny bug in GET endpoint for the client collection pertaining to await

<div>
client home page timeline
Screen Shot 2023-03-19 at 5 44 36 PM
</div>

## PATCH Client Timeline Sample Curl
curl --location --request PATCH 'http://localhost:3001/api/client/64177ef559295797ab75b5cb/timeline' \
--header 'Content-Type: application/json' \
--data '{
    "text": "wee!",
    "category": "UPDATES",
    "title": "Test updates",
    "timestamp": "2023-3-29"
}'
## PATCH Client Timeline Sample Response
Successfully updated the timeline field of client with id 64177ef559295797ab75b5cb

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [ ] I have run `npm run lint:fix`
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (if required)
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works (Delete if frontend)

### CODE REVIEW

File: client/src/components/Timeline/ActivityTimeline.tsx [hunk 1]
```diff
@@ -80,7 +84,7 @@ export default function ActivityTimeline({
 
   //sends request to backend to add a new note to the timeline for the given employer
   async function updateTimeline(timelineObject: object) {
-    fetch(`/api/employer/${employerId}/timeline`, {
+    fetch(`/api/${type}/${objectId}/timeline`, {
```
Comment (kenneth-miura) [inline client/src/components/Timeline/ActivityTimeline.tsx:87]: GODLIKE REFACTORING

File: client/src/components/Timeline/DisplayTimelinePosts.tsx [hunk 2]
```diff
@@ -29,9 +30,9 @@ export default function DisplayTimelinePosts({id}: {id: string}) {
   const filteredPosts = timelinePosts.filter(post => {
     // Show all posts
     if (value === 0) {
-      return true;
+      return tabs.some(tab => post.category === tab);
```
Comment (kenneth-miura) [inline client/src/components/Timeline/DisplayTimelinePosts.tsx:33]: oh is this b/c the options of tabs are different depending on if the timeline is in client or employer?

**Review by kenneth-miura — APPROVED** [review]
common only child W

**Review by ganeshasapu — APPROVED** [review]
all of this lgtm! great job! :)

**Review by jordanjanakievski — APPROVED** [review]
Everything looks good to me. This PR will be unaffected by my removal of the contacts endpoints since it will be handled on the Employer page.

File: client/src/components/Timeline/ActivityTimeline.tsx (hunk 1 above)
Comment (edwardhuahan) [reply client/src/components/Timeline/ActivityTimeline.tsx:87]: thats goated

**Comment by edwardhuahan** [comment]
I like how the code is very neat and extendable,  👍

**Pull request merged**